spatialos-sdk/src/worker/snapshot.rs

```diff
@@ -5,12 +5,39 @@ use crate::{
 use spatialos_sdk_sys::worker::*;
 use std::{ffi::CString, path::Path};
 
+#[derive(Debug)]
+pub enum SnapshotError {
+    Bad,
+    InvalidData,
+    EOF,
+}
+
+#[derive(Debug)]
+pub struct SnapshotErrorState {
+    pub error: SnapshotError,
+    pub message: String,
```

Comment:
Unsure if its more idiomatic to wrap this string into each of the error codes 🤔

Reply:
Yeah, it probably is... I'll change it.